### ISSUE TITLE
sin or cos of matrix crashes 

### ISSUE THREAD

**Issue by daniepf** [JuliaLang/julia#29040]
sin(A), cos(A), log(A), but not exp(A), crash Julia when A is a matrix.

**Comment by JeffBezanson** [comment]
Can you post an example?

**Comment by JeffBezanson** [comment]
Also please provide the output of `versioninfo()`.

**Comment by adamslc** [comment]
For some reason, compilation takes a ridiculously long time in this case:
```julia
julia> versioninfo()
Julia Version 1.0.0
Commit 5d4eaca (2018-08-08 20:58 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU           X5650  @ 2.67GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, westmere)
Environment:
  JULIA_PKGDIR = /scratch/ladams/julia

julia> A = rand(3,3);

julia> @time sin(A)
  4.516312 seconds (7.34 M allocations: 348.534 MiB, 7.93% gc time)
3×3 Array{Float64,2}:
 0.26415    0.142019  0.271248
 0.477338   0.72241   0.145316
 0.0544759  0.130292  0.700738
```

**Comment by daniepf** [comment]
_       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.1.0-DEV.160 (2018-09-02)
 _/ |\__'_|_|_|\__'_|  |  Commit f77c36bc7* (2 days old master)
|__/                   |

julia> A = rand(2,2)
2×2 Array{Float64,2}:
 0.545976  0.501158
 0.374316  0.52999 

julia> exp(A)
2×2 Array{Float64,2}:
 1.88988   0.885354
 0.661273  1.86164 

julia> sin.(A)
2×2 Array{Float64,2}:
 0.519253  0.480442
 0.365636  0.505525

julia> sin(A)
Illegal inttoptr
          %11 = ptrtoint { double, double } addrspace(13)* %10 to i64
Illegal inttoptr
          %scevgep13 = ptrtoint double addrspace(13)* %scevgep to i64

signal (6): Aborted
in expression starting at no file:0
gsignal at /usr/bin/../lib64/libc.so.6 (unknown line)
abort at /usr/bin/../lib64/libc.so.6 (unknown line)
unknown function (ip: 0x7f70f381c224)
_ZN4llvm13FPPassManager13runOnFunctionERNS_8FunctionE at /usr/bin/../lib64/libLLVM-6.0.so (unknown line)
_ZN4llvm13FPPassManager11runOnModuleERNS_6ModuleE at /usr/bin/../lib64/libLLVM-6.0.so (unknown line)
_ZN4llvm6legacy15PassManagerImpl3runERNS_6ModuleE at /usr/bin/../lib64/libLLVM-6.0.so (unknown line)
unknown function (ip: 0x7f70f38fe9c1)
unknown function (ip: 0x7f70f390230a)
unknown function (ip: 0x7f70f3903b39)
unknown function (ip: 0x7f70f38a5e78)
unknown function (ip: 0x7f70f38f813d)
jl_fptr_trampoline at /usr/bin/../lib64/libjulia.so.1 (unknown line)
jl_apply_generic at /usr/bin/../lib64/libjulia.so.1 (unknown line)
unknown function (ip: 0x7f70f3989ae4)
unknown function (ip: 0x7f70f3989763)
unknown function (ip: 0x7f70f398a897)
unknown function (ip: 0x7f70f398af9e)
unknown function (ip: 0xfffffffffffffffe)
unknown function (ip: 0x7f70d6cb1dbf)
unknown function (ip: 0xffffffffffffffff)
unknown function (ip: 0x7f70f398b4bf)
unknown function (ip: 0x7f70f3862974)
jl_toplevel_eval_in at /usr/bin/../lib64/libjulia.so.1 (unknown line)
unknown function (ip: 0x7f70d95d5851)
jl_apply_generic at /usr/bin/../lib64/libjulia.so.1 (unknown line)
unknown function (ip: 0x7f70d9c484a2)
unknown function (ip: 0x7f70d9c48703)
jl_apply_generic at /usr/bin/../lib64/libjulia.so.1 (unknown line)
unknown function (ip: 0x7f70f3849a21)
unknown function (ip: 0xffffffffffffffff)
Allocations: 11208366 (Pool: 11204386; Big: 3980); GC: 23


Warning: Program '/usr/bin/julia' crashed.

**Comment by daniepf** [comment]
Done with latest Julia version on Fedora 28  intel 64bit

**Comment by daniepf** [comment]
julia> versioninfo()
Julia Version 1.1.0-DEV.160
Commit f77c36bc7* (2018-09-02 11:27 UTC)
Platform Info:
  OS: Linux (x86_64-redhat-linux)
  CPU: Intel(R) Core(TM) i7-4800MQ CPU @ 2.70GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, haswell)

**Comment by yuyichao** [comment]
This seems to be system llvm, is it patched?

**Comment by daniepf** [comment]
No the llvm rpms are :  
llvm-libs-6.0.1-6.fc28.x86_64
llvm5.0-libs-5.0.1-7.fc28.x86_64

**Comment by JeffBezanson** [comment]
Yes, this looks like something that would happen without our LLVM patches.

**Comment by yuyichao** [comment]
Unmatched llvm is not supported so you either need to stop using system llvm or apply all the needed patches.....

**Comment by daniepf** [comment]
The julia rpm comes from 
https://copr.fedorainfracloud.org/coprs/nalimilan/julia-nightlies/packages/
where llvm has not been updated since 3 years...

**Comment by ViralBShah** [comment]
@nalimilan keeps updated Julia dependencies I believe.

**Comment by daniepf** [comment]
Replacing the livm-libs from Fedora by the libvm6-libs from nalimilan/julia-nightlies  solved the issue. 

julia> A = rand(2,2)
2×2 Array{Float64,2}:
 0.457956  0.638364
 0.611146  0.3163  

julia> sin(A)
2×2 Array{Float64,2}:
 0.366715  0.552955
 0.529379  0.244013

**Comment by daniepf** [comment]
Thanks!

**Comment by nalimilan** [comment]
I don't understand how it's possible that you installed the julia package without llvm6, since that's one of its dependencies.

**Comment by daniepf** [comment]
Fedora updated non patched  libvm to 6.0.1 but not nalimilan up to now.

**Comment by nalimilan** [comment]
OK, I've found the source of the bug. The julia package depended on libLLVM-6.0.so, but on Fedora 28 that dependency is also provided by llvm-libs, without the patches needed by Julia. I've pushed a new build which explicitly requires the patched llvm6, and ensures Julia loads that even when llvm-libs is installed.
